airbyte-integrations/connectors/source-surveymonkey/unit_tests/unit_test.py

```diff
@@ -0,0 +1,56 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+import json
+
+import pendulum
+import vcr
+from airbyte_cdk.models import SyncMode
+from source_surveymonkey import SourceSurveymonkey
+
+
+def test_get_updated_state():
+    source = SourceSurveymonkey()
```

Comment:
this test is currently an integration test because it reads data from survey monkey directly. Instead it should just be a unit test. Can't we create an instance of the streams and then directly test the `get_updated_state` method e.g: 

```
stream = IncrementalSurveyMonkeyStream()

record = {'field1':"value", "cursor_field": "2021-01-01"}
current_state = {"cursor_field": "2021-02-01"}
expected_state = current_state

assert stream.get_updated_state(record, current) == expected_state
```

then `record`, `current_state`, and `expected_state` can all be parametrized via pytest to repeat this test for many input values

Reply:
added this (date comparison test), but instantiate from `Surveys` stream that completely inherits `get_updated_state` method. Because IncrementalSurveyMonkeyStream has undefined abstract properties `path` and `cursor_field` and cannot be instantiated. Also I kept previous test in the file